mmdet/datasets/dataset_wrappers.py

```diff
@@ -401,15 +401,26 @@ def __getitem__(self, idx):
                 continue
 
             if hasattr(transform, 'get_indexes'):
-                indexes = transform.get_indexes(self.dataset)
-                if not isinstance(indexes, collections.abc.Sequence):
-                    indexes = [indexes]
-                mix_results = [
-                    copy.deepcopy(self.dataset[index]) for index in indexes
-                ]
-                results['mix_results'] = mix_results
-
-            results = transform(results)
+                while True:
```

Comment:
Need to add a limitation to these two loops to avoid infinite loop.

Reply:
And raise some detailed error messages to tell users their data processing pipeline may be incorrect.